clients/python/generate_schema_types.py

```diff
@@ -0,0 +1,471 @@
+#!/usr/bin/env python3
+"""
+Generate Python dataclasses from JSON schemas derived from Rust types.
+
+
+For pure value types (mostly used in APIs), we use `schemars` to generate the JSON schemas from Rust types,
+and `datamodel-code-generator` to generate the Python dataclasses from the JSON schemas.
+
+Rust Types (with annotations)
+    ↓ (cargo test export_schema)
+JSON Schemas (in clients/schemas/)
+    ↓ (python generate_schema_types.py)
+Python Dataclasses (in clients/python/tensorzero/generated_types)
+
+
+This script runs both steps of the generation pipeline:
+
+1. Runs cargo tests to generate JSON schemas from Rust types (using #[export_schema])
+    - See `tensorzero-derive` crate for more details on the macro, and https://graham.cool/schemars/ for schemars docs.
+2. Generates Python dataclasses from those JSON schemas
+    - For detailed configuration options, see https://koxudaxi.github.io/datamodel-code-generator/.
+
+It also sets up customizations to make them work well together:
+
+1.  `schemars` generates one JSON Schema file for each Rust type, bringing in all its dependencies. Naively
+    generating Python dataclasses from these JSON Schemas would cause dependencies to be repeatedly generated, so
+    we merge them into a single schema file before generating Python dataclasses.
+2.  To generate the __init__.py file so clients can simply write `from tensorzero.generated_types import X`, we
+    collect all the type names from the schema and export them.
+3.  We generate a custom header file for the generated types to include the `UNSET` sentinel value.
```

Comment:
add something for rewrite_refs_recursive

Reply:
Done.